fe/fe-core/src/main/java/org/apache/doris/analysis/MultiPartitionDesc.java

```diff
@@ -0,0 +1,316 @@
+    public static final String MONTHS_FORMAT = "yyyyMM";
+    public static final String MONTH_FORMAT = "yyyy-MM";
+    public static final String YEAR_FORMAT = "yyyy";
+    public static final String DATETIME_FORMAT = "yyyy-MM-dd HH:mm:ss";
+
+
+
+    private final String partitionPrefix = "p_";
+    private LocalDateTime startTime;
+    private LocalDateTime endTime;
+
+    private DateTimeFormatter startDateTimeFormat;
+    private DateTimeFormatter endDateTimeFormat;
+
+
+    private Long timeInterval;
+    private final PartitionKeyDesc partitionKeyDesc;
+    private TimestampArithmeticExpr.TimeUnit timeUnitType;
+    private final Map<String, String> properties;
+    private final List<SinglePartitionDesc> singlePartitionDescList = Lists.newArrayList();
+
+    private final ImmutableSet<TimestampArithmeticExpr.TimeUnit> timeUnitTypeMultiPartition = ImmutableSet.of(
+            TimestampArithmeticExpr.TimeUnit.HOUR,
+            TimestampArithmeticExpr.TimeUnit.DAY,
+            TimestampArithmeticExpr.TimeUnit.WEEK,
+            TimestampArithmeticExpr.TimeUnit.MONTH,
+            TimestampArithmeticExpr.TimeUnit.YEAR
+    );
+
+    private final Integer maxAllowedLimit = Config.max_multi_partition_num;
+
+    public MultiPartitionDesc(PartitionKeyDesc partitionKeyDesc,
+                          Map<String, String> properties) throws AnalysisException {
+        this.partitionKeyDesc = partitionKeyDesc;
+        this.properties = properties;
+        this.timeIntervalTrans();
+        this.timeTrans();
+    }
+
+    public List<SinglePartitionDesc> getSinglePartitionDescList() throws AnalysisException {
+        if (singlePartitionDescList.size() == 0) {
+            buildMultiPartitionToSinglePartitionDescs();
+        }
+        return singlePartitionDescList;
+    }
+
+    private List<SinglePartitionDesc> buildMultiPartitionToSinglePartitionDescs() throws AnalysisException {
+        String partitionName;
+        long countNum = 0;
+        int dayOfWeek = 1;
+        int dayOfMonth = 1;
+        String partitionPrefix = this.partitionPrefix;
+        LocalDateTime startTime = this.startTime;
+        if (properties != null) {
+            if (properties.containsKey(DynamicPartitionProperty.START_DAY_OF_WEEK)) {
+                String dayOfWeekStr = properties.get(DynamicPartitionProperty.START_DAY_OF_WEEK);
+                try {
+                    DynamicPartitionUtil.checkStartDayOfWeek(dayOfWeekStr);
+                } catch (DdlException e) {
+                    throw new AnalysisException(e.getMessage());
+                }
+                dayOfWeek = Integer.parseInt(dayOfWeekStr);
+            }
+            if (properties.containsKey(DynamicPartitionProperty.START_DAY_OF_MONTH)) {
+                String dayOfMonthStr = properties.get(DynamicPartitionProperty.START_DAY_OF_MONTH);
+                try {
+                    DynamicPartitionUtil.checkStartDayOfMonth(dayOfMonthStr);
```

Comment:
If there are some methods in `DynamicPartitionUtil` that can be used in this class,
then I think these methods should be moved to a common class, may be `TimeUtils.java`

Reply:
This piece of code is to solve the conflict between dynamic partitioning and multi partitioning at one create table sql. It is reasonable to refer to the params/methods  of dynamic partitions for judgment. These params/methods are still belong to dynamic partitions essentially, so I think they should  not be moved.